Marlin/src/feature/powerloss.cpp

```diff
@@ -224,15 +225,108 @@ void PrintJobRecovery::save(const bool force/*=false*/) {
 #if PIN_EXISTS(POWER_LOSS)
 
   void PrintJobRecovery::_outage() {
+    #if ENABLED(BACKUP_POWER_SUPPLY_PARK_WAITING)
+      // Timer
+      static millis_t power_loss_delay = 0;
+      const millis_t ms = millis();
+      if (!power_loss_delay) power_loss_delay = ms + BACKUP_POWER_SUPPLY_DELAY_BEFORE;
+      if (PENDING(ms, power_loss_delay)) return;
+      power_loss_delay = 0;
+    #endif
```

Comment:
I moved this delay first because it was blocked by the `lock` test. Seems sensible here.